tests/core/test_snapshot.py

```diff
@@ -1957,10 +1960,30 @@ def _loader(snapshot_ids: t.Set[SnapshotId]) -> t.Collection[Snapshot]:
     assert cached_snapshot.model._query_renderer._optimized_cache is not None
     assert cached_snapshot.model._data_hash is not None
     assert cached_snapshot.model._metadata_hash is not None
+    assert not cached_snapshot.intervals
+    assert not cached_snapshot.dev_intervals
 
     cache.clear()
     assert cache.get_or_load([snapshot.snapshot_id], _loader) == (
         {snapshot.snapshot_id: snapshot},
         set(),
     )
     assert loader_called_times == 2
+
+
+def test_snapshot_pickle_intervals(make_snapshot):
+    snapshot = make_snapshot(
+        SqlModel(
+            name="test_model_name",
+            query=parse_one("SELECT 1"),
+        )
+    )
+    snapshot.add_interval("2023-01-01", "2023-01-02")
+    snapshot.add_interval("2023-01-01", "2023-01-02", is_dev=True)
+
+    assert len(snapshot.intervals) > 0
+    assert len(snapshot.dev_intervals) > 0
+
+    loaded_snapshot = pickle.loads(pickle.dumps(snapshot))
```

Comment:
you can also assert that the snapshot obj hasn't lost it's intervals

Reply:
added